### DISCUSSION TITLE
Is there a way to use tailwind.config.ts instead?

### DISCUSSION THREAD

**Discussion by finnmerlett** [tailwindlabs/tailwindcss#6839] [Help]
I want to keep chunks of my theme in ts files elsewhere in my project. I want to be able to access them directly with type hinting but also want to be able to import them into my tailwind config file (at the moment you can only 'require' CommonJs files into the tailwind.config.js file). I wouldn't even need it to be typescript necessarily (though that would be nice), I just want to be able to import in es6 modules style. Is there any way to set this up?

**Comment by aranoe** [comment]
+1

**Comment by lightyen** [comment]
Use swc or esbuild to achieve this:

```js
// tailwind.config.js

const { register } = require("@swc-node/register/register")
const { readDefaultTsConfig } = require("@swc-node/register/read-default-tsconfig")
const path = require("path")
register(readDefaultTsConfig(path.join(__dirname, "tsconfig.json")))
module.exports = require("./tailwind.config.ts").default
```

```ts
// tailwind.config.ts

export default {
  plugins: [],
  theme: {
    extend: {},
  },
} // as import("tailwindcss/types").Config

```

**Reply by bluefire2121** [reply]
tailwind.config.ts 

```
import type { Config } from "tailwindcss"

export default {
  plugins: [],
  theme: {
    extend: {},
  },
} satisfies Config
```
